### PR TITLE
Add new functionality for corporation tax page

### PR DESCRIPTION
**Pull request by drosser**
Resolves: SFA-549.

### CODE REVIEW

File: src/main/java/uk/gov/companieshouse/web/accounts/controller/corporationtax/CorporationTaxController.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    @PostMapping
+    public String postCorporationTax(@PathVariable String companyNumber,
+                                     @ModelAttribute("corporationTax") @Valid CorporationTax corporationTax,
+                                     BindingResult bindingResult) {
+
+        if (bindingResult.hasErrors()) {
+            return getTemplateName();
+        }
+
+        if (corporationTax.getFileChoice().equals(true)) {
```
Comment (BSaunders95) [inline src/main/java/uk/gov/companieshouse/web/accounts/controller/corporationtax/CorporationTaxController.java]: `.equals` not required here; could just use `if (corporationTax.getFileChoice())`. Could the name of the boolean on the model be more descriptive, like `fileWithHMRC`?

File: src/main/java/uk/gov/companieshouse/web/accounts/controller/corporationtax/CorporationTaxController.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+            return "redirect:" + "https://www.gov.uk/file-your-company-accounts-and-tax-return";
```
Comment (BSaunders95) [inline src/main/java/uk/gov/companieshouse/web/accounts/controller/corporationtax/CorporationTaxController.java]: We've tried to use the `UrlBasedViewResolver.REDIRECT_URL_PREFIX` constant in place of raw 'redirect:' strings

**Comment by BSaunders95** [comment]
There are no accompanying unit tests to the new controller class

File: src/main/resources/templates/corporationtax/corporationTax.html [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+                <div class="govuk-grid-column-two-thirds">
+                    <form id="corporationTax-form" class="form" th:action="@{''}" th:object="${corporationTax}" method="POST">
+                        <div th:replace="fragments/globalErrors :: globalErrors"></div>
+
+                        <div class="govuk-form-group" th:classappend="${#fields.hasErrors('fileChoice')} ? 'govuk-form-group--error' : ''">
+                            <div th:replace="fragments/inlineYesNoRadioButtonsNoTextBox :: inlineYesNoRadioButtonsNoTextBox (
+                                  legendText = 'Do you also want to file your corporation tax with HMRC now?',
+                                  radioButtonsFieldName = 'fileChoice',
+                                  )">
+
```
Comment (Mathsh) [inline src/main/resources/templates/corporationtax/corporationTax.html]: minor space

**Pull request merged**